In [1]:
import numpy as np

def fred(a,b):
    c = ( a + b ) * 12
    return c

a = 13.2
b = 3.9
c = fred(a,b)
print ("c = ",c)

c =  205.2


In [2]:
def rule3(dA,dB):
    dQ = np.sqrt(dA**2+dB**2)
    return dQ

dA = 1.4
dB = 2.1

dQ = rule3(dA,dB)
print (dQ)

2.5238858928247927


In [4]:
x = np.array([1, 1.1,1.2,0.9,0.85])
y = x*3
print (y)
ysq = y**2
print (ysq)

err_x = np.std(x)/np.sqrt(5)
avgx = np.average(x)
print(avgx,err_x)

[3.   3.3  3.6  2.7  2.55]
[ 9.     10.89   12.96    7.29    6.5025]
1.01 0.05727128425310541


$\delta F = g\delta m$

$\delta Q = \sqrt{(\delta A)^2 + (\delta B)^2}$

$\frac{\delta A}{A}$

# Rule 1 Function

In [6]:
def rule1(c,dA):
    dQ = c*dA
    return dQ

c = 3
dA = 7

dQ = rule1(c,dA)
print (dQ)

21


# Rule 2 Function

In [12]:
def rule2(c,m,A,dA):
    dQ = c * m * ((A)**m-1) * dA
    return dQ

c = 2
m = -5
A = 2
dA = 4

dQ = rule2(c,m,A,dA)
print (dQ)

38.75


# Rule 3 Function

In [13]:
def rule3(dA,dB):
    dQ = np.sqrt(dA**2+dB**2)
    return dQ

dA = 4.7
dB = 9.8

dQ = rule3(dA,dB)
print (dQ)

10.868762579061151


# Rule 4 Function

In [22]:
def rule4(Q,A,m,dA,B,n,dB):
    dQ = Q*np.sqrt((((m*dA)/A)**2)+((n*dB)/(b))**2)
    return dQ

Q = 3.53
A = 1.588
m = 1
dA = .005
B = .989
n = -.5
dB = .001

dQ = rule4(Q,A,m,dA,B,n,dB)
print (dQ)

0.011123819496927314


###### This is the same error in velocity as I found in my calculations.

# 

In [24]:
x = np.array ([1.1,1.3,1.4,0.9,0.95,1.05])
avgx = np.average(x)
err_x = np.std(x)/np.sqrt(6)

print(avgx,err_x)

1.1166666666666667 0.073282810879294


##### The average of these values is equal to 1.12 respectively and the error in x is equat to 0.07 respectively.

# Rule 3 Equation

$\delta Q = \sqrt{(\delta A)^2 + (\delta B)^2}$

# Rule 4 Equation

$\frac{\delta Q}{Q} = \sqrt{(\frac{m*(\delta A)}{A})^2 + (\frac{n*(\delta B)}{B})^2}$